pkg/gui/controllers/branches_controller.go

```diff
@@ -209,6 +214,30 @@ func (self *BranchesController) copyPullRequestURL() error {
 	return nil
 }
 
+func (self *BranchesController) copyBranchURL() error {
+	branch := self.context().GetSelected()
+
+	branchExistsOnRemote := self.git.Remote.CheckRemoteBranchExists(branch.Name)
```

Comment:
This checks only for the hardcoded upstream of `origin`, look [here](https://github.com/ryand67/lazygit/blob/099297b7f7fdf844d5554118586e1beadea7a82b/pkg/commands/git_commands/remote.go#L50):
```
// CheckRemoteBranchExists Returns remote branch
func (self *RemoteCommands) CheckRemoteBranchExists(branchName string) bool {
	_, err := self.cmd.
		New(
			fmt.Sprintf("git show-ref --verify -- refs/remotes/origin/%s",
				self.cmd.Quote(branchName),
			),
		).
		DontLog().
		RunWithOutput()

	return err == nil
}
```

I propose we send the `branch.UpstreamRemote` as a parameter alongside `branchName`.

Reply:
Sorry, proposing that upstreamRemote gets passed in with the CheckRemoteBranchExists? Or with the getBranchURL method? Think I'm confused as to where that becomes helpful. The rest of the tweaks are finished, gonna wait for clarification on this one :)

Reply:
> Sorry, proposing that upstreamRemote gets passed in with the CheckRemoteBranchExists?

Yes.

> Think I'm confused as to where that becomes helpful.

Try adding a new remote. It's not gonna be `refs/remotes/origin/<something>`, but it's gonna be e.g. `refs/remotes/ryans-fork/<something>`. But `CheckRemoteBranchExists` has `origin` hardcoded in its `git` command, how's that gonna work for something _not on origin_? :)